db/migrate/20240423155817_change_workflow_execution_state_to_enum.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+# Migration to update state column in workflow executions from type string to int to utilize enums
+class ChangeWorkflowExecutionStateToEnum < ActiveRecord::Migration[7.1]
+  def up # rubocop:disable Metrics/MethodLength
+    rename_column :workflow_executions, :state, :old_state
+    add_column :workflow_executions, :state, :integer
```

Comment:
**nitpick (blocking):** Can you please add `,null: false, default: 0` here, so that `state` will never be nil.

Reply:
Added in 8f20f7b3cbb9b0ed878962908a1ad6d4431dbda3. Fixed associated test including removing explicit `state = :initial/new` assignments